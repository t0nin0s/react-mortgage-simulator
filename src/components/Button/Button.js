import React from 'react'
import styles from './button.css'
import FontAwesome from 'react-fontawesome'
import PropTypes from 'prop-types'

const MyButton = (props) => {
  return (
    <button
      className={styles.button}
      id={props.id}
      value={props.value}
      onClick={props.handleOnChange}>
      <span className={styles.button__text}>
        {props.text}
      </span>
      <span className={styles.button__icon}>
        <FontAwesome
          name={props.icon}
          size='3x'
          style={{ color: '#A06FEFs'}}
        />
      </span>
    </button>
  )
}

MyButton.propTypes = {
  id: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleOnChange: PropTypes.func.isRequired
}

export default MyButton
