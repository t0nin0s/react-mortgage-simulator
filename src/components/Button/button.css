:local(.button) {
  width: 100%;
  color: #A06FEF;
  font-weight: bold;
  background-color: white;
  border: 1px solid #ccc;
  height: 5rem;
  text-align: left;
  padding-left: 3rem;
  border-radius: 2rem;
  -moz-border-radius: 2rem;
  -webkit-border-radius: 2rem;
  margin: 0.5rem 0;
}

:local(.button--selected) {
  color: red;
}

:local(.button__text) {
  float:left;
  font-weight:bold;
  line-height: 4rem;
}

:local(.button__icon) {
  height:4rem;
  display: inline;
  vertical-align: middle;
  float: right;
  padding-right: 1rem;
}
