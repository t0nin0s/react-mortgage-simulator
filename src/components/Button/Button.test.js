import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from './Button'

describe('SignupFormPersonal renders as expected', () => {
  it('should render as expected', () => {
    const component = shallow(
      <Button
        id=''
        value=''
        icon=''
        handleOnChange={jest.fn()} />
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
