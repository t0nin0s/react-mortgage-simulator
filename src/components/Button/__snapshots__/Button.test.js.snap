// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupFormPersonal renders as expected should render as expected 1`] = `
<button
  className=" "
  id=""
  onClick={[Function]}
  value=""
>
  <span />
  <span>
    <FontAwesome
      name=""
      size="3x"
      style={
        Object {
          "color": undefined,
        }
      }
      value=""
    />
  </span>
</button>
`;
